### PR TITLE
[FIX] port base_gs1_barcode to v7

### PR DESCRIPTION
**Pull request by clonedagain**
Make base_gs1_barcode installable.
Relicence as AGPL-3.
Match coding conventions more closely
Use a distinct exception. This exception is placed in the package's scope to make it easier to use from other modules. Now use "from base_gs1_barcode import InvalidGS1Barcode"

### CODE REVIEW

File: __unported__/base_gs1_barcode/__init__.py [hunk 1]
```diff
@@ -1,22 +1,25 @@
 # -*- coding: utf-8 -*-
 ##############################################################################
 #
-#    This module is copyright (C) 2012 Numérigraphe SARL. All Rights Reserved.
+#    This module is copyright (C) 2014 Numérigraphe SARL. All Rights Reserved.
```
Comment (pedrobaeza) [inline __unported__/base_gs1_barcode/__init__.py]: You should put 2012-2014

Comment (pedrobaeza) [inline __unported__/base_gs1_barcode/__init__.py]: All Rights Reserved is not valid anymore

File: __unported__/base_gs1_barcode/__init__.py [hunk 2]
```diff
@@ -1,22 +1,25 @@
...
+#    You should have received a copy of the GNU Affero General Public License
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ##############################################################################
 
 from . import gs1_barcode
 from . import res_users
+
+# Bring the main exception into the package's scope for easier reuse
+from .exceptions import InvalidGS1Barcode
```
Comment (pedrobaeza) [inline __unported__/base_gs1_barcode/__init__.py]: Why do you need this here?

File: __unported__/base_gs1_barcode/__openerp__.py [hunk 3]
```diff
@@ -1,19 +1,19 @@
 # -*- coding: utf-8 -*-
 ##############################################################################
 #
-#    This module is copyright (C) 2012 Numérigraphe SARL. All Rights Reserved.
+#    This module is copyright (C) 2014 Numérigraphe SARL. All Rights Reserved.
```
Comment (pedrobaeza) [inline __unported__/base_gs1_barcode/__openerp__.py]: Same remarks

File: __unported__/base_gs1_barcode/gs1_barcode.py [hunk 4]
```diff
@@ -1,19 +1,19 @@
 # -*- coding: utf-8 -*-
 ##############################################################################
 #
-#    This module is copyright (C) 2012 Numérigraphe SARL. All Rights Reserved.
+#    This module is copyright (C) 2014 Numérigraphe SARL. All Rights Reserved.
```
Comment (pedrobaeza) [inline __unported__/base_gs1_barcode/gs1_barcode.py]: Same remarks

File: __unported__/base_gs1_barcode/gs1_barcode.py [hunk 5]
```diff
@@ -24,44 +24,42 @@
...
-                                  help='Indicates whether a digit is expected '
-                                       'before the data for this Application '
-                                       'Identifier to indicate the position of '
-                                       'the decimal point.'
-                                  ),
-        'type': fields.selection([ ('string', 'Any character string'),
-                                    ('numeric', 'Numeric value'),
-                                    ('date', 'Date') ],
-                                    'Data Type', required=1),
+        'ai': fields.char(
```
Comment (pedrobaeza) [inline __unported__/base_gs1_barcode/gs1_barcode.py:37]: Label is missing (Application Identifer)

File: __unported__/base_gs1_barcode/__init__.py (hunk 2 above)
Comment (clonedagain) [inline __unported__/base_gs1_barcode/__init__.py]: So I can simply import it in other modules using "from adons.base_gs1_barcode import InvalidGS1Barcode"

Comment (pedrobaeza) [inline __unported__/base_gs1_barcode/__init__.py]: Thanks for the explanation.

**Comment by clonedagain** [comment]
@pedrobaeza all should be fixed.

File: base_gs1_barcode/res_users.py [hunk 6]
```diff
@@ -1,44 +1,42 @@
...
 
     _columns = {
-        # XXX those should be properties, not standard fields
         'gs1_barcode_prefix': fields.char(
             'Prefix', size=64,
             help="The prefix that the barcode scanner will send when GS1-128 "
                  "or GS1-Datamatrix codes are scanned. No prefix is expected "
-                 "if this fields is left empty"
-            ),
+                 "if this fields is left empty"),
```
Comment (pedrobaeza) [inline base_gs1_barcode/res_users.py]: s/fields/field

File: base_gs1_barcode/res_users.py [hunk 7]
```diff
@@ -1,44 +1,42 @@
...
+#    You should have received a copy of the GNU Affero General Public License
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ##############################################################################
 
 from openerp.osv import orm, fields
 
+
 # XXX This would probably be best in res_config_users
 class res_users(orm.Model):
```
Comment (pedrobaeza) [inline base_gs1_barcode/res_users.py]: ResUsers

File: base_gs1_barcode/res_users.py [hunk 8]
```diff
@@ -1,44 +1,42 @@
...
     """Add the bar code decoding configuration to the user profile"""
     _inherit = 'res.users'
 
     _columns = {
-        # XXX those should be properties, not standard fields
         'gs1_barcode_prefix': fields.char(
             'Prefix', size=64,
```
Comment (pedrobaeza) [inline base_gs1_barcode/res_users.py:31]: Why is this at user level and not at company level?

**Comment by pedrobaeza** [comment]
Please update also POT file.

File: base_gs1_barcode/res_users.py (hunk 8 above)
Comment (clonedagain) [inline base_gs1_barcode/res_users.py:31]: Because this needs to be configured the same as the on on the barcode reader.

**Comment by clonedagain** [comment]
By the way I have the feeling that part of this could best served as python library, but I'm clueless about how to make it properly. If anyone has pointers to share I'll be glad to follow them (at least when I'll migrate to v8).

**Comment by pedrobaeza** [comment]
Yeah, it could be, or make it as a kind of extensible framework for adding any kind of barcode, like we did on bank import modules from bank-statement-reconcile repo.

**Comment by clonedagain** [comment]
Translation template added.

**Comment by pedrobaeza** [comment]
Some minor issues but :+1:

**Comment by clonedagain** [comment]
@pedrobaeza I didn't notice your later comments, I think it's fixed now.

**Comment by pedrobaeza** [comment]
Thanks

File: base_gs1_barcode/i18n/fr.po [hunk 9]
```diff
@@ -0,0 +1,829 @@
...
+"Language-Team: \n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: \n"
+"Plural-Forms: \n"
+
+#. module: base_gs1_barcode
+#: model:gs1_barcode,name:base_gs1_barcode.AREA
+msgid "Product Area, in square meters"
+msgstr "Product Area, in square meters"
```
Comment (yvaucher) [inline base_gs1_barcode/i18n/fr.po]: Please keep empty the translations when not translated

**Comment by yvaucher** [comment]
I guess this has been in prod for some times @clonedagain ?

If you can empty the wrong french translantions I will proceed with merging

**Comment by gurneyalex** [comment]
@clonedagain could you rebase the PR to trigger a new build?

**Comment by clonedagain** [comment]
@yvaucher @gurneyalex it's done, let's see what Travis thinks of it.

**Comment by yvaucher** [comment]
@clonedagain Thanks, travis was fast, I think it only checked for PEP8 and lint.

Could it be that TESTS="1" is missing in travis config?

**Comment by gurneyalex** [comment]
@clonedagain https://github.com/numerigraphe/stock-logistics-barcode/pull/1 should make runbot green

**Comment by gurneyalex** [comment]
travis issue should be fixed by #13

**Comment by clonedagain** [comment]
I think it's fixed

**Comment by yvaucher** [comment]
@clonedagain Was it intended to add all that text in commit message  7063a04 ?

Otherwise travis is green :+1:

**Comment by clonedagain** [comment]
Yes I thought it illustrated the problem well, but I can cut it if it's 
a problem, just tell me.

**Comment by clonedagain** [comment]
There was another view with a `type`, this new commit should fix it.

**Comment by yvaucher** [comment]
@clonedagain Thanks

**Comment by StefanRijnhart** [comment]
@clonedagain I was not aware of this PR, and I migrated the module to 8.0. See https://github.com/OCA/stock-logistics-barcode/pull/15. There are two details that I found were not as per specifications. Can you confirm those?

At the same time, I switched the custom exception to the conventional ValidationError, now available in openerp.exceptions.

I also modified the prefix and the separator to global configuration, but I understand now that that was a mistake. Will change that back.

**Comment by StefanRijnhart** [comment]
Nice module, BTW!

**Comment by StefanRijnhart** [comment]
:+1:, code review.

**Pull request merged**